### PR TITLE
Rearrange files so that one can git clone directly into the extension folder

### PR DESCRIPTION
**Pull request by davidanthoff**
This should make it easier to work on this (once there are more extension points exposed).

### CODE REVIEW

**Comment by mikewl** [comment]
Thanks. I think the November update will contain more extension types. Once the vscode package manager is introduced I will change this into a proper package. That may change the folder structure again.

Other than those mentioned in the Julia-users discussion, are there any other features you can think of that would be useful? Please open issues if you think of any. Tonight I will bring over the list I made as well.

**Comment by davidanthoff** [comment]
Sounds good. I was actually surprised that the tmLanguage file is just dropped into the extension folder, I had expected that this yo program would convert it into some VS Code specific format (like what they seem to use for the other languages in the extensions folder in the main folder of the program). 

My main question right now is whether they will enable enough flexibility in their plugin model to enable things like inline plotting (a la Juno) etc.

**Comment by mikewl** [comment]
I just looked around at the C# support and feel that inline plotting will definitely be possible. With the C# support there is already small windows popping up which can be closed after which should allow for inline results and plots as well as other functionality. It really just depends on them allowing it. It may not be possible in the beginning but once VSCode extension functionality is improved upon I feel it will be there.

It would be nice if the extensions would allow things like the debug tab holding variable information for example or we could make our own tab. If a Julia specific tab is possible then it could do things like package management and show current variables and other information. Something like a MATLAB environment.

Their native descriptor looks quite good actually. I read through the one for python and wondered if that could not be adapted for Julia. I wonder whether the yeoman generator would not convert it at some point.

In short: I think inline results should be possible. The support is there for working with a console as well. All that's needed is for the VSCode devs to allow us to use that functionality.

**Comment by mikewl** [comment]
Opened a few issues now which should cover most desired functionality. Add more or comment on some if you have any suggestions!
